runtime/autoload/man.vim

```diff
@@ -49,48 +46,54 @@ function! man#open_page(count, count1, ...) abort
       " user explicitly set a count
       let sect = string(a:count)
     endif
-    let [sect, name] = s:verify_exists(sect, name)
+    let [sect, name, path] = s:verify_exists(sect, name)
   catch
     call s:error(v:exception)
     return
   endtry
   call s:push_tag()
   let bufname = 'man://'.name.(empty(sect)?'':'('.sect.')')
-  let found_man = s:find_man()
-  if getbufvar(bufname, 'manwidth') ==# s:manwidth()
-    if found_man
-      silent execute 'buf' bufnr(bufname)
-    else
-      execute 'split' bufname
+  if a:mods !~# 'tab' && s:find_man()
+    if s:manwidth() ==# getbufvar(bufname, 'manwidth')
+      silent execute 'buf' bufname
+      call man#set_window_local_options()
+      keepjumps 1
+      return
     endif
-    keepjumps 1
+    noautocmd execute 'edit' bufname
```

Comment:
It is used so that the extraction and verification isn't repeated in `man#read_page` when the BufReadCmd autocmd is triggered. Yes, they will be respected because the filetype is set later in `s:read_page`.